model/src/resource/device.rs

```diff
@@ -1,15 +1,27 @@
 use std::fmt;
 
-use nmos_schema::is_04;
+use nmos_schema::is_04::{v1_0_x, v1_3_x};
 use serde::Serialize;
+use serde_json::json;
 use uuid::Uuid;
 
 use crate::{
     resource::Node,
-    version::{is_04::V1_0, APIVersion},
+    version::{is_04::V1_0, is_04::V1_3, APIVersion},
 };
 
-use super::{ResourceCore, ResourceCoreBuilder};
+use super::{Registerable, ResourceCore, ResourceCoreBuilder};
+
+macro_rules! registration_request {
```

Comment:
Since the different API versions are created by schemafy in nmos-rs we can switch which crate we use to create the JSON  using this macro. Could be a lot cleaner, but this way we keep the option of implementing different API versions in the future open without too much implementation effort